pymeasure/instruments/fwbell/fwbell5080.py

```diff
@@ -39,83 +40,77 @@ class FWBell5080(Instrument):
         meter = FWBell5080('/dev/ttyUSB0')  # Connects over serial port /dev/ttyUSB0 (Linux)
 
         meter.units = 'gauss'               # Sets the measurement units to Gauss
-        meter.range = 3e3                   # Sets the range to 3 kG
+        meter.range = 1                     # Sets the range to 3 kG
         print(meter.field)                  # Reads and prints a field measurement in G
 
         fields = meter.fields(100)          # Samples 100 field measurements
         print(fields.mean(), fields.std())  # Prints the mean and standard deviation of the samples
 
     """
 
-    id = Instrument.measurement(
-        "*IDN?", """ Reads the idenfitication information. """
-    )
     field = Instrument.measurement(
-        ":MEAS:FLUX?",
+        ":MEASure:FLUX?",
         """ Reads a floating point value of the field in the appropriate units.
         """,
-        get_process=lambda v: v.split(' ')[0]  # Remove units
+        # Remove units
+        get_process=lambda v: float(v.replace('T', '').replace('G', '').replace('Am', ''))
+
     )
     UNITS = {
         'gauss': 'DC:GAUSS', 'gauss ac': 'AC:GAUSS',
         'tesla': 'DC:TESLA', 'tesla ac': 'AC:TESLA',
         'amp-meter': 'DC:AM', 'amp-meter ac': 'AC:AM'
     }
     units = Instrument.control(
-        # TODO a newer version expects ':UNIT:FLUX:%s', is this right?
-        ":UNIT:FLUX?", ":UNIT:FLUX%s",
+        ":UNIT:FLUX?", ":UNIT:FLUX:%s",
         """ A string property that controls the field units, which can take the
         values: 'gauss', 'gauss ac', 'tesla', 'tesla ac', 'amp-meter', and
         'amp-meter ac'. The AC versions configure the instrument to measure AC.
         """,
         validator=strict_discrete_set,
         values=UNITS,
-        map_values=True,
-        get_process=lambda v: v.replace(' ', ':')  # Make output consistent with input
+        map_values=True
+    )
+
+    range = Instrument.control(
+        ":SENS:FLUX:RANG?", ":SENS:FLUX:RANG %d",
+        """ An integer property that controls the maximum field range in the active units.
+        Value is dependent on the currently selected units (gauss, tesla, amp-meter)
+
+        +--------+--------+---------+-----------+
+        | Value  | gauss  |  tesla  | amp-meter |
+        +--------+--------+---------+-----------+
+        | 0      | 300 G  |  30  mT | 23.88 kAm |
+        +--------+--------+---------+-----------+
+        | 1      | 3 kG   |  300 mT | 238.8 kAm |
+        +--------+--------+---------+-----------+
+        | 2      | 30 kG  |  3 T    | 2388  kAm |
+        +--------+--------+---------+-----------+
+        """,
+        validator=strict_discrete_set,
+        values=[0, 1, 2],
+        cast=int
     )
 
     def __init__(self, adapter, **kwargs):
```

Comment:
(minor thing) Can you move the __init__ to the top of the class, please?

Reply:
I would be more than happy to. There isn't consistency across the code base, but because `pymeasure.instruments.keithley import Keithley2400` is prominent in the documentation quick start I have taken instrument coding style clues from that. And that instrument has `def __init__()` at start of method section halfway through. 

I think `def __init__()` at the beginning makes a lot more sense.

Reply:
Move init in last commit.

Reply:
Yeah, we only recently formalised that (still in a documentation PR), but have been nudging it in that direction for a while.